core_api/wasm-test/commit/src/lib.rs

```diff
@@ -70,25 +60,25 @@ fn hc_commit(data: *mut c_char, entry_type_name: &str, entry_content : &str)
 // Utils
 //-------------------------------------------------------------------------------------------------
 
-// Convert json input into something meaningful
-fn deserialize<'s, T: Deserialize<'s>>(data: *mut c_char) -> T {
-    let c_str = unsafe { CStr::from_ptr(data) };
-    let actual_str = c_str.to_str().unwrap();
+// Convert json data in a memory buffer into a meaningful data struct
+fn deserialize<'s, T: Deserialize<'s>>(ptr_data: *mut c_char) -> T {
+    let ptr_safe_c_str = unsafe { CStr::from_ptr(ptr_data) };
+    let actual_str = ptr_safe_c_str.to_str().unwrap();
     serde_json::from_str(actual_str).unwrap()
 }
 
-// Convert a data struct into json memory buffer
-fn serialize<T: Serialize>(data: *mut c_char, internal: T) -> i32 {
-    let json = serde_json::to_string(&internal).unwrap();
-    let bytes = json.as_bytes();
-    let len = bytes.len();
-    let mem = unsafe { slice::from_raw_parts_mut(data, len) };
+// Write a data struct into a memory buffer as json string
+fn serialize<T: Serialize>(ptr_data: *mut c_char, internal: T) -> i32 {
+    let json_str = serde_json::to_string(&internal).unwrap();
+    let json_bytes = json_str.as_bytes();
```

Comment:
https://docs.serde.rs/serde_json/fn.to_vec.html
This should do the trick

Reply:
done